packages/clay-core/src/drop-down/Item.tsx

```diff
@@ -48,7 +48,13 @@ export const Item = React.forwardRef<HTMLLIElement, Props>(function ItemInner(
 	const As = href ? Link : onClick ? 'button' : 'span';
 
 	return (
-		<li data-index={dataIndex} ref={ref} role="presentation" style={style}>
+		<li
+			data-index={dataIndex}
+			onClick={onClick}
```

Comment:
In fact you can move this to the inner element, it's better to keep the `onClick` on the button.

Reply:
Thanks @matuzalemsteles! 
I'll make this change and send the fix in a few minutes.